mono/mini/interp/interp.c

```diff
@@ -5978,14 +5974,14 @@ interp_exec_method_full (InterpFrame *frame, ThreadContext *context, FrameClause
 				}
 			}
 
-			frame->endfinally_ip = NULL;
+			endfinally_ip = NULL;
 
-			if (old_list != frame->finally_ips && frame->finally_ips) {
-				ip = (const guint16*)frame->finally_ips->data;
-				frame->finally_ips = g_slist_remove (frame->finally_ips, ip);
+			if (old_list != finally_ips && finally_ips) {
+				ip = (const guint16*)finally_ips->data;
+				finally_ips = g_slist_remove (finally_ips, ip);
 				sp = frame->stack; /* spec says stack should be empty at endfinally so it should be at the start too */
 				vt_sp = (unsigned char *) sp + imethod->stack_size;
-				MINT_IN_DISPATCH (*ip);
+				goto main_loop;
```

Comment:
The goto seems to help the compiler also.
See https://github.com/mono/mono/pull/16306.

Reply:
Should add comments on gotos.